index.d.ts

```diff
@@ -4,6 +4,6 @@ interface Options {
   limit: number;
 }
 
-declare function safeRegex<T>(regex: RegExp, opts?: Options | T): void;
+declare function safeRegex<T>(regex: RegExp | string | T, options?: Options): boolean;
```

Comment:
- The union for the regex with RegExp and string  is an improvement 👍 
- Could you explain how the generic (the `<T>` and the `T` in union of the first parameter) relates to the original function? The declaration without it (`export function safeRegex (regex: RegExp | string, options?: Options): boolean`) seems to match the original function just as well, but I might've missed some thing.